pkg/security/x509.go

```diff
@@ -350,3 +373,14 @@ func ParseTenantURISAN(rawURL string) (roachpb.TenantID, string, error) {
 	}
 	return roachpb.MustMakeTenantID(tID), username, nil
 }
+
+// ParseTenantNameURISAN extracts the user and tenant name contained within a tenant URI SAN.
+func ParseTenantNameURISAN(rawURL string) (roachpb.TenantName, string, error) {
+	uri := strings.Replace(rawURL, tenantURISANPrefixString, "", 1)
+	parts := strings.Split(uri, "/")
```

Comment:
I think it is _OK_ because we control the caller. But this function is only correct if you've already checked that the URL is in fact a tenant-name URI SAN. Otherwise, since `strings.Replace` doesn't error if it doesn't actually replace anything, it seems like one could construct a string that could pass through this function despite not being a `crdb://` uri.